### DISCUSSION TITLE
DevExpress Report

### DISCUSSION THREAD

**Discussion by EtherGhost** [AvaloniaUI/Avalonia#6182] [Q&A] [answered]
What it the preferred way to use DevX Report with Avalonia, windows and Linux, ASP WebApi .NET5?

Is there a PDF viewer for Avalonia?

**Comment by npolyak** [comment]
as far as I know, neither DevEx nor Telerik are not compatible with Avalonia - at least not yet. If someone knows anything otherwise, please tell us.

**Reply by EtherGhost** [reply]
Yes I  think you are right but it should be possible to let the reports to be created on the WebApi and then sent to the user as pdf. Or something ...

**Comment by EtherGhost** [comment, marked as answer]
It's working just fine by creating a DevX report in PDF format on a ASP.NET5 WebApi and returning that file from the API.

Links:
https://stackoverflow.com/questions/36042614/how-to-return-a-pdf-from-a-web-api-application#36043779

https://codeburst.io/download-files-using-web-api-ae1d1025f0a9

https://docs.devexpress.com/XtraReports/14950/detailed-guide-to-devexpress-reporting/store-and-distribute-reports/print-and-export-reports-without-a-preview

**Comment by dennis-garavsky** [comment]
You can find a GitHub example that shows how to use [DevExpress Reports](https://www.devexpress.com/subscriptions/reporting/) with Avalonia at https://github.com/DevExpress-Examples/reporting-avalonia-get-started
